pkg/kube/client.go

```diff
@@ -1,15 +1,78 @@
 package kube
 
 import (
+	"errors"
 	"os"
 
+	"k8s.io/client-go/tools/clientcmd"
+
+	appsclient "github.com/openshift/client-go/apps/clientset/versioned"
+	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
 )
 
+// Client struct exposes interfaces for kubernetes as well as openshift if available
+type Clients struct {
+	KubernetesClient    kubernetes.Interface
+	OpenshiftAppsClient appsclient.Interface
+	IsOpenshift         bool
+}
+
+// GetClients returns a `Clients` object containing both openshift and kubernetes clients with an openshift identifier
+func GetClients() Clients {
+	client, err := GetClient()
+	isOpenshift := true
+	if err != nil {
+		logrus.Fatalf("Unable to create Kubernetes client error = %v", err)
+	}
+
+	appsClient, err := GetOpenshiftAppsClient()
+	if err != nil {
+		logrus.Warnf("Unable to create Openshift Apps client error = %v", err)
+		isOpenshift = false
+	}
+	return Clients{
+		KubernetesClient:    client,
+		OpenshiftAppsClient: appsClient,
+		IsOpenshift:         isOpenshift,
+	}
+}
+
+func isOpenshift() bool {
+	client, err := GetClient()
+	if err != nil {
+		logrus.Fatalf("Unable to create Kubernetes client error = %v", err)
+	}
+	_, err = client.RESTClient().Get().AbsPath("/apis/project.openshift.io").Do().Raw()
+	if err == nil {
+		return true
+	}
+	return false
+}
+
+// GetOpenshiftAppsClient returns an Openshift Client that can query on Apps
+func GetOpenshiftAppsClient() (*appsclient.Clientset, error) {
+	if !isOpenshift() {
+		return nil, errors.New("Not running on Openshift")
```

Comment:
While testing Reloader image `stakater/reloader:SNAPSHOT-PR-74-10` on `minikube`. I am getting a log multiple times(>15) when Reloader is started. 

```bash
time="2019-07-05T09:14:05Z" level=warning msg="Unable to create Openshift Apps client error = Not running on Openshift"
```

My suggestions:

1- log level should be `info` instead of `warning`.
2- Either reloader detect in which environment it is running or a flag should specify its environment.

Reply:
I have a flag IsOpenshift in Clients struct which has been used everywhere. I the isOpenshift method shouldn't be called multiple times. I will see why and reduce the function call to just 1 and change it to info with a more meaningful message

Reply:
Reloader generates multiple logs only for the first time when it starts. After that when a resource changes it generates a single log.

Reply:
Clients are required at multiple places and the isOpenshift method was getting invoked multiple times. I am moving it out as a package variable which only gets computed once and then the functions can use that variable instead of calling isOpenshift again

Reply:
@aliartiza75 can you please check the latest image again?

Reply:
@waseem-h sure, once done I will let you know.